MSAL/src/instance/MSALDeviceInfoProvider.m

```diff
@@ -41,10 +41,27 @@ @implementation MSALDeviceInfoProvider
 - (void)deviceInfoWithRequestParameters:(MSIDRequestParameters *)requestParameters
                         completionBlock:(MSALDeviceInformationCompletionBlock)completionBlock
 {
+    void (^deviceInfoCompletion)(MSALDeviceInformation *, NSError *) = ^void(MSALDeviceInformation *msalDeviceInfo, NSError *error)
+    {
+        if (!msalDeviceInfo)
+        {
+            msalDeviceInfo = [MSALDeviceInformation new];
+            msalDeviceInfo.deviceMode = MSALDeviceModeDefault;
+        }
+
+        NSDictionary *deviceRegMetaDataInfo = [MSIDWorkPlaceJoinUtil getRegisteredDeviceMetadataInformation:nil];
+        if (deviceRegMetaDataInfo)
+        {
+            [msalDeviceInfo addRegisteredDeviceMetadataInformation:deviceRegMetaDataInfo];
+        }
+    
+        completionBlock(msalDeviceInfo, error);
```

Comment:
You shouldn't return both object and the error.

Reply:
Option -1: Will be very straight forward and will returns registration state without disturbing deviceState and SSO extension.

Option-2: Mostly we have seen failures from line#98 i.e. MSIDSSOExtensionGetDeviceInfoRequest::executeRequestWithCompletion returns error so if we go with return msalDeviceInfo = nil, error. then its as if we are not adding any value and API will failed to return registration state due to SSO extension failures.

Or can we add some enum or scope so that caller will explicitly mention for WPJ state?

Reply:
@valangot, @jasoncoolmax, could you discuss and conclude?

Reply:
Discussed offline with @jasoncoolmax, returning devcieInfo and error from couple of places, We will monitor errors and AppCenter Defender telemetry and try to resolve SSO requests intermittent issue.

cc @oldalton

Reply:
@valangot, I've dismissed my review, so feel free to merge once you get an approval from @jasoncoolmax